python/caffe/coord_map.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import division
+import numpy as np
+from caffe import layers as L
+
+PASS_THROUGH_LAYERS = ['AbsVal', 'ReLU', 'PReLU', 'Dropout', 'LRN', 'Eltwise',
+        'BatchNorm', 'BNLL', 'Log', 'Exp', 'MVN', 'Power', 'Sigmoid', 'Split',
+        'TanH', 'Threshold']
+
+def conv_params(fn):
+    params = fn.params.get('convolution_param', fn.params)
```

Comment:
Not quite; perhaps this warrants a comment. What's going on here is that the parameters are normally read from `fn.params`, which contains the conv params for a conv layer and the pooling params for a pooling layer. However, for layers which _share_ the `ConvolutionParameter` message type (i.e., currently only deconv layer), we need to explicitly ask for `convolution_param`, since `Convolution` is not the name of the layer.